instrumentation/kotlinx-coroutines/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kotlinxcoroutines/instrumentationannotations/WithSpanInstrumentation.java

```diff
@@ -0,0 +1,718 @@
+              scopeLocal = newLocal(Type.getType(Scope.class));
+
+              visitLabel(start);
+            }
+
+            @Override
+            public void visitMaxs(int maxStack, int maxLocals) {
+              visitLabel(handler);
+              visitTryCatchBlock(start, handler, handler, null);
+              super.visitMaxs(maxStack, maxLocals);
+            }
+
+            @Override
+            public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
+              AnnotationVisitor annotationVisitor = super.visitAnnotation(descriptor, visible);
+              // remember value and kind from the @WithSpan annotation
+              if ("Lapplication/io/opentelemetry/instrumentation/annotations/WithSpan;"
+                  .equals(descriptor)) {
+                return new AnnotationVisitor(api, annotationVisitor) {
+                  @Override
+                  public void visit(String name, Object value) {
+                    if ("value".equals(name) && value instanceof String) {
+                      withSpanValue = (String) value;
+                    }
+                    super.visit(name, value);
+                  }
+
+                  @Override
+                  public void visitEnum(String name, String descriptor, String value) {
+                    if ("kind".equals(name)
+                        && "Lapplication/io/opentelemetry/api/trace/SpanKind;".equals(descriptor)) {
+                      spanKind = value;
+                    }
+                    super.visitEnum(name, descriptor, value);
+                  }
+                };
+              }
+              return annotationVisitor;
+            }
+
+            @Override
+            public void visitEnd() {
+              super.visitEnd();
+
+              // If a suspend method does not contain any blocking operations or has no code after
+              // the blocking operation it gets compiled to a regular method that we instrument the
+              // same way as the regular @WithSpan handling does. We create the span at the start of
+              // the method and end it in before every return instruction and in exception handler.
+              // If a suspend method has a blocking operation and code that needs to be executed
+              // after it, we start the span only when the coroutine was started, on resume we just
+              // activate the scope. We end the span when coroutine completes, otherwise we only
+              // close the scope.
+              // First we'll search for a bytecode sequence that looks like
+              // 64: aload         6
+              // 66: getfield      #444                // Field
+              // io/opentelemetry/javaagent/instrumentation/kotlinxcoroutines/KotlinCoroutinesInstrumentationTest$b2$1.label:I
+              // 69: tableswitch   { // 0 to 1
+              //                0: 92
+              //                1: 181
+              //          default: 210
```

Comment:
Is this bytecode sequence present in the same method that is annotated with `@WithSpan`? Do you have some examples of `javap -v` bytecode we could match to the instrumentation here?
I tried to decompile the test code but while I did find this code pattern, I couldn't understand how it connects to the `@WithSpan` method

Reply:
Nevermind, I forgot about `-p` 🤦